components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,356 @@
+import {
+  Component, Input, OnChanges, ViewChild, ElementRef, SimpleChanges, OnInit
+} from '@angular/core';
+import * as d3 from 'd3';
+import {
+  DayPercentage
+} from 'app/entities/desktop/desktop.model';
+
+
+@Component({
+  selector: 'app-simple-line-graph',
+  templateUrl: './simple-line-graph.component.html',
+  styleUrls: ['./simple-line-graph.component.scss']
+})
+
+export class SimpleLineGraphComponent implements OnChanges, OnInit {
+
+  // labels too high
+  // disappearing render
+
+  constructor(
+    private chart: ElementRef
+  ) { }
+
+  @ViewChild('svg', { static: true }) svg: ElementRef;
+  @Input() data: DayPercentage[] = [];
+  @Input() width = 900;
+  @Input() height = 156; // we want a 116px height on the ticks, this minus margins
+  private locked: number = null;
```

Comment:
A number is a primitive type; you should only store numbers in it, so initialize to zero here rather than null. (In most languages this would not even be valid; I am surprised it is in TypeScript.)
```suggestion
  private locked: number = 0;
```

Reply:
thanks for pointing this out - It made me go back and realize this whole time i was and should have been returning a string.